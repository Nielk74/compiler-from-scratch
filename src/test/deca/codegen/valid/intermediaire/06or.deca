{
    int x = 1;
    int j = 2;
    
    if (x != j || x > x){
        println(x, " != ", j, " || ", x, " > ", x);
    }

    if (x >= 5 || x < j){
        println(x, " >= ", 5, " || ", x, " < ", j);
    }

    if (!(x > j || x == 40)){
        println("!(", x, " > ", j, " || ", x, " == ", 40, ")");
    }

    if (!!(x > j || 0 <= j)){
        println("!!(", x, " > ", j, " || ", 0, " <= ", j, ")");
    }
}