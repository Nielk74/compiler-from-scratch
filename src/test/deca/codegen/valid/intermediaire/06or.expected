1 != 2 || 1 > 1
1 >= 5 || 1 < 2
!(1 > 2 || 1 == 40)
!!(1 > 2 || 0 <= 2)
